lib/socketclusterclient/reconnect.rb

```diff
@@ -66,7 +41,7 @@ def reconnect
   #
   # @return [Boolean] Attempts finished
   #
-  def reconnection_attempts_finished
-    @attempts_made == @max_attempts
+  def should_reconnect
+    @enable_reconnection && (@max_attempts.nil? || (@attempts_made < @max_attempts))
```

Comment:
can u put comment above this explaining the logic, would be easier to understand

Reply:
If automatic reconnection is enabled and connect automatically and also attempts made is less than  max attempts, allow reconnection to server.